src/rust/src/wallet.rs

```diff
@@ -170,6 +270,14 @@ impl Wallet {
             .collect::<Vec<_>>();
         let mut result = vec![];
         for (action_idx, ivk, note, recipient, memo) in bundle.decrypt_outputs_for_keys(&keys) {
+            // Generate the nullifier for the received (unspent) note and add it to
+            // the nullifiers map so that we can detect when the note is later spent.
+            if let Some(fvk) = self.key_store.viewing_keys.get(&ivk) {
+                self.nullifiers
+                    .entry(note.nullifier(fvk))
+                    .or_insert_with(BTreeSet::new);
+            }
+
             // Mark the witness tree with the fact that we want to be able to compute
             // a witness for this note.
             let note_data = DecryptedNote { note, memo };
```

Comment:
The blank line 284 makes it look like the comment only applies to line 283, which is not the case. I think it applies to what are currently lines 283 and 285.

Reply:
That comment is actually completely wrong. Will fix it.